www/artworks/new.php

```diff
@@ -64,20 +80,42 @@
 				<div>
 					<label>
 						Artwork Name
-						<input type="text" name="artwork-name" required="required"/>
+						<input type="text" name="artwork-name" required="required"
+						       value="<?= $artwork->Name ?>"/>
 					</label>
 					<label for="artwork-year">
 						Year of Completion
 						<label>
 							(circa?
-							<input type="checkbox" name="artwork-year-is-circa"/>)
+							<input
+								type="checkbox"
+								name="artwork-year-is-circa"
+								value="<?= $artwork->CompletedYearIsCirca ?>"
+							/>)
 						</label>
-						<input type="number" id="artwork-year" name="artwork-year" min="0" max="<?= gmdate('Y') ?>"/>
+						<input
+							type="number"
+							id="artwork-year"
+							name="artwork-year"
+							min="0"
+							max="<?= gmdate('Y') ?>"
+							value="<?= $artwork->CompletedYear ?>"
+						/>
 					</label>
 				</div>
 				<label>
 					Artwork Tags
-					<input type="text" name="artwork-tags" placeholder="tags, comma-separated"/>
+					<input
+						type="text"
+						name="artwork-tags"
+						placeholder="tags, comma-separated"
+						value="<?php
+							$tags = $artwork->ArtworkTags ?? [];
+							$tags = array_column($tags, 'Name');
+							$tags = implode(', ', $tags);
```

Comment:
You could bundle up these three lines into a new method or property in `Artwork.php`. That way you could replace the logic here with something like:

```php
$artwork->ArrayTagsImploded
```

or

```php
$artwork->ArrayTagsString 
```